embassy-stm32/src/clock.rs

```diff
@@ -255,7 +255,7 @@ impl TimerInner {
         self.stop_and_reset();
 
         let psc = timer_freq.0 / TICKS_PER_SECOND as u32 - 1;
-        let psc: u16 = psc.try_into().unwrap();
+        let psc: u16 = unwrap!(psc.try_into().map_err(|_| "psc division"));
```

Comment:
This is a new change. Unfortunately, the `map_err` is necessary because the error type here doesn't have a `Format` instance.